File: soyeong/programmers-2/모음사전.py
answer = 0
def DFS(string, word):
    alphabets = ['A', 'E', 'I', 'O', 'U']
    global answer
    answer += 1

    # 단어를 찾았으면 return
    if string == word:
        return True

    # word 길이는 최대 5개 이므로 5개 길이가 됬는데 단어가 일치하지 않으면 false
    if len(string) == 5:
        return False

    for a in alphabets:
        if (DFS(string + a, word) == True):  # 알파벳하나를 더 붙여 재귀
            return True

def solution(word):
    global answer
    alphabets = ['A', 'E', 'I', 'O', 'U']

    for a in alphabets:
        if (DFS(a, word) == True):
            return answer

File: kyulee/programmers-2/모음사전.java
public class 모음사전 {
    class Solution {
        public int solution(String word) {
            int answer = 0;
            int[] weights = {781, 156, 31, 6, 1}; // 각 자리수의 가중치

            for (int i = 0; i < word.length(); i++) {
                char c = word.charAt(i);
                int index = "AEIOU".indexOf(c); // 각 자리의 알파벳이 몇 번째 알파벳인지

                answer += weights[i] * index + 1; // 해당 자리수의 가중치와 알파벳의 순서를 곱하여 더함
            }

            return answer;
        }
    }
}
